common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/OAuth2Strategy.java

```diff
@@ -105,59 +115,93 @@ protected HttpResponse performTokenRequest(final GenericTokenRequest request) th
 
 
     /**
-     * Construct the authorization endpoint URI based on issuer and path to the authorization endpoint
+     * Construct the authorization endpoint URI based on issuer and path to the authorization endpoint.
      * NOTE: We could look at basing this on the contents returned from the OpenID Configuration document
      *
-     * @return
+     * @return URI
      */
     protected Uri createAuthorizationUri() {
         //final Uri.Builder builder = new Uri.Builder().scheme(originalAuthority.getProtocol()).authority(host).appendPath(path);
         Uri authorizationUri = Uri.withAppendedPath(mIssuer, mAuthorizationEndpoint);
         return authorizationUri;
     }
 
+    protected String getAuthorizationEndpoint() {
+        return mAuthorizationEndpoint;
+    }
+
+    protected void setAuthorizationEndpoint(String authorizationEndpoint) {
+        this.mAuthorizationEndpoint = authorizationEndpoint;
+    }
+
+    protected String getTokenEndpoint() {
+        return mTokenEndpoint;
+    }
+
+    protected void setTokenEndpoint(String tokenEndpoint) {
+        this.mTokenEndpoint = tokenEndpoint;
+    }
+
```

Comment:
Could mark params final but I'm not going to stress it since these might change.